src/o365/graph/commands/schemaextension/schemaextension-add.ts

```diff
@@ -0,0 +1,244 @@
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  id: string;
+  description: string;
+  owner: string;
+  targetTypes: string;
+  properties: string;
+}
+
+class GraphSchemaExtensionAdd extends GraphCommand {
+  public get name(): string {
+    return `${commands.SCHEMAEXTENSION_ADD}`;
+  }
+
+  public get description(): string {
+    return 'Creates a Microsoft Graph schema extension';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.id = args.options.id;
+    telemetryProps.description = args.options.description;
+    telemetryProps.owner = args.options.owner;
+    telemetryProps.targetTypes = args.options.targetTypes;
+    telemetryProps.properties = args.options.properties;
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    auth
+      .ensureAccessToken(auth.service.resource, cmd, this.debug)
+      .then((): request.RequestPromise => {
+        if (this.verbose) {
+          cmd.log(`Adding schema extension with id '${args.options.id}'...`);
+        }
+
+        const targetTypes = args.options.targetTypes.split(',').map(t => t.trim());
+        let properties: any = null;
+        properties = JSON.parse(args.options.properties);
+
+        const requestOptions: any = {
+          url: `${auth.service.resource}/v1.0/schemaExtensions`,
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${auth.service.accessToken}`,
+            accept: 'application/json;odata.metadata=none',
+            'content-type': 'application/json'
+          }),
+          body: {
+            id: args.options.id,
+            description: args.options.description,
+            owner: args.options.owner,
+            targetTypes,
+            properties
+          },
+          json: true
+        };
+
+        if (this.debug) {
+          cmd.log('Executing web request...');
+          cmd.log(requestOptions);
+          cmd.log('');
+        }
+
+        return request.post(requestOptions);
+      })
+      .then((res: any): void => {
+        if (this.debug) {
+          cmd.log('Response:')
+          cmd.log(res);
+          cmd.log('');
+        }
+
+        cmd.log(`Schema extension added with Id '${args.options.id}'`);
+
+        if (this.verbose) {
+          cmd.log(vorpal.chalk.green('DONE'));
+        }
+
+        cb();
+      }, (err: any) => this.handleRejectedODataJsonPromise(err, cmd, cb));
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-i, --id <id>',
+        description: `The unique identifier for the schema extension definition`
+      },
+      {
+        option: '-d, --description [description]',
+        description: 'Description of the schema extension'
+      },
+      {
+        option: '-o, --owner <owner>',
```

Comment:
`-o` is already used by the `output` option, so we can't use it here

Reply:
Should we use another short arg, or not at all ?

Reply:
Removed short for owner argument